### PR TITLE
[WIP] Adding more detail to the v1alpha3 config godocs

### PR DESCRIPTION
**Pull request by alejandrox1**
This adds more details to the available fields and settings in the
kind.sigs.k8s.io/v1alpha3 config.

Signed-off-by: alejandrox1 <alarcj137@gmail.com>

### CODE REVIEW

**Comment by alejandrox1** [comment]
Godoc does not build for branches but if anyone is interested in reviewing the changes locally you can try by running:
```
$ godoc -http=localhost:6060
```

and going to http://localhost:6060/pkg/sigs.k8s.io/kind/pkg/apis/config/v1alpha3/

File: pkg/apis/config/v1alpha3/doc.go [hunk 1]
```diff
@@ -19,4 +19,70 @@ limitations under the License.
...
+// Basics
+//
+// The preferred way to configure kind is to pass an YAML configuration file with the --config option.
+//
+// kind supports the following configuration types:
+//
+//     apiVersion: kind.sigs.k8s.io/v1alpha3
+//     kind: Cluster
+//
+// The Cluster configuration type should be used to configure kind-cluster settings,
```
Comment (neolit123) [inline pkg/apis/config/v1alpha3/doc.go]: kind-cluster -> kind cluster

File: pkg/apis/config/v1alpha3/doc.go [hunk 2]
```diff
@@ -19,4 +19,70 @@ limitations under the License.
...
+//
+// - Networking contains cluster wide network settings.
+//
+// - Nodes contains the list of nodes defined in the kind cluster
+// If unset this will default to a single control-plane node.
+// Note that if more than one control plane is specified, an external
+// control plane load balancer will be provisioned implicitly.
+//
+// Here is a fully populated example of a single YAML file containing multiple
+// configuration types to be used during a `kind create cluster` run.
```
Comment (neolit123) [inline pkg/apis/config/v1alpha3/doc.go]: should the "kind-example-config" from here, link to this new godocs page now?
https://kind.sigs.k8s.io/docs/user/quick-start/#configuring-your-kind-cluster

Comment (alejandrox1) [reply pkg/apis/config/v1alpha3/doc.go]: Ah this is actually something that I wanted to run by you all 1) do we want a godocs page that goes into detail about the kind config or 2) add all this info to https://kind.sigs.k8s.io/docs/ ?

Option 1 is the "standard", most projects (that I've seen) that have a config link to their godocs for full details and there you can browse between the different versions. I.e., https://godoc.org/k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm#pkg-subdirectories

Option 2, we could put all these docs in the site itself, may be even dedicate an entire page for the config itself and give multiple config examples.

What do you think?
/cc @BenTheElder @munnerz @tao12345666333 @aojea

Comment (alejandrox1) [reply pkg/apis/config/v1alpha3/doc.go]: ping @BenTheElder what do you think? is godoc worth it or should we stick to the official docs for this?

Comment (BenTheElder) [reply pkg/apis/config/v1alpha3/doc.go]: I think we should follow up on changing the link, but imo getting godoc or something similar that covers the full type spec would be best long term

Comment (tao12345666333) [reply pkg/apis/config/v1alpha3/doc.go]: IMO, kind is both a cli tool and a library, so maybe we should using option 1.

**Comment by BenTheElder** [comment]
Doing a configuration docs page now, and moving on to newer API versions

Thank you for your effort on this

**Pull request closed without merging**